crates/rpc-types/src/transaction/request.rs

```diff
@@ -90,6 +90,29 @@ impl OpTransactionRequest {
         self
     }
 
+    /// Check this builder's preferred type, based on the fields that are set.
+    ///
+    /// Types are preferred as follows:
+    /// - Deposit Transactions if `chain_id` is not set
+    /// - EIP-7702 if authorization_list is set
+    /// - EIP-4844 if sidecar or max_blob_fee_per_gas is set
+    /// - EIP-2930 if access_list is set
+    /// - Legacy if gas_price is set and access_list is unset
+    /// - EIP-1559 in all other cases
```

Comment:
```suggestion
    /// - See [`TransactionRequest::preferred_type`].
```

I just delegate docs when I delegate a call, that way I know I don't create doc bugs in future